asv_bench/benchmarks/period.py

```diff
@@ -137,7 +64,68 @@ def time_asfreq():
         self.per.asfreq('A')
 
 
-class period_standard_indexing(object):
+class PeriodIndexConstructor(object):
+    goal_time = 0.2
+
+    params = ['D']
+    param_names = ['freq']
+
+    def setup(self, freq):
+        self.freq = freq
+        self.rng = date_range('1985', periods=1000)
+        self.rng2 = date_range('1985', periods=1000).to_pydatetime()
+
+    def time_from_date_range(self):
+        PeriodIndex(self.rng, freq=self.freq)
+
+    def time_from_pydatetime(self):
+        PeriodIndex(self.rng2, freq=self.freq)
+
+
+class DataFramePeriodColumn(object):
+    goal_time = 0.2
+
+    def setup_cache(self):
+        rng = pd.period_range(start='1/1/1990', freq='S', periods=20000)
+        df = pd.DataFrame(index=range(len(rng)))
+        return rng, df
+
+    def time_setitem_period_column(self, tup):
+        rng, df = tup
+        df['col'] = rng
+
+
+class PeriodIndexAlgorithms(object):
```

Comment:
you could also param this one (I would just call it Algorithms, the 'period' is already in the file name) for ['series', 'index']

Reply:
I thought about that but decided against it because the two cases treat the data differently, would need a `if cls is Series: ...` in the setup method.  But heck, let's go for it.

Reply:
I would rather do a `data = ...; if typ == 'series': data = pd.Series(data)`